txnkv/transaction/2pc.go

```diff
@@ -1608,8 +1608,32 @@ func (c *twoPhaseCommitter) cleanup(ctx context.Context) {
 		cleanupKeysCtx := context.WithValue(c.store.Ctx(), retry.TxnStartKey, ctx.Value(retry.TxnStartKey))
 		var err error
 		if c.txn.IsPipelined() {
-			// TODO: cleanup pipelined txn
-			// TODO: broadcast txn status
+			if len(c.pipelinedCommitInfo.pipelinedStart) != 0 && len(c.pipelinedCommitInfo.pipelinedEnd) != 0 {
```

Comment:
Should we close the TTL manager here?

Reply:
It's closed here.
https://github.com/tikv/client-go/blob/70049ae310bfcc02bbd8f3929ec4fd779d697e7f/txnkv/transaction/txn.go#L720